utils/state-machine.js

```diff
@@ -254,27 +265,55 @@ export const reservationMachine = createMachine(
       },
       lastDive: {
         on: {
-          [STATE_ACTIONS.DIVER_INFORMATION]: {
-            target: "diverInformation",
+          [STATE_ACTIONS.EQUIPMENT]: {
+            target: "equipment",
             actions: ["pushToPreviousState", "setLastDive"],
           },
-
           //Back
           [STATE_ACTIONS.DEEPEST_DIVE]: {
             target: "deepestDive",
             actions: ["popFromPreviousState", "removeDeepestDive"],
           },
         },
       },
+      equipment: {
+        on: {
+          [STATE_ACTIONS.DIVER_INFORMATION]: {
+            target: "diverInformation",
+            actions: ["pushToPreviousState", "setEquipment"],
+          },
+
+          [STATE_ACTIONS.FIN_SIZE]: {
```

Comment:
I don't think we shouuld have it this way, ideally equipment + fin size is one state